src/js/kits/model-kit-select/model-kit-select.js

```diff
@@ -0,0 +1,152 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { ModelKitConsumer } from '../../data/kits/modelkit';
+import { BrandSelect } from './brand-select';
+import { ScaleSelect } from './scale-select';
+import { SearchInput } from './search-input';
+import { KitPreview } from './kit-preview';
+
+const DEBOUNCE = 300;  // debounce in ms
+
+const modelKitConsumer = new ModelKitConsumer();
+
+// TODO: pass selected kit IDs as prop
+// TODO: handle toggle changes & allowMultiple yes/no
+
+const ModelKitSelect = (props) => {
+    const {
+        label,
+        htmlName,
+        allowMultiple,
+        selected,
+    } = props;
+
+    // track search parameters state
+    const [searchParams, setSearchParams] = useState({});
```

Comment:
I'd look into using `useReducer` hook for this instead, which is better for handling more complex state. That way all that `updateSearchParam` logic could be abstracted into reducer and simplified.

Reply:
Definitely an improvement. Shame it doesn't track key updates and you still have to return a new object, but oh well, it's complex to track that stuff.